code/gui5.py

```diff
@@ -0,0 +1,204 @@
+        else:
+            self.constrain_pcs = False
+
+        do_clustering(self.input_file_stem,self.output_file,self.v_scalar,self.constrain_partition,self.constrain_pcs)
+
+# Setting up the root of the window
+window = tk.Tk()
+window.title("Distributional Learning")
+#window.columnconfigure(0,minsize=400,weight=1)
+#window.rowconfigure(0,weight=1)
+
+################### Vector Model Builder ################
+vmb_frame = tk.Frame(master=window)
+vmb_frame.grid(row=0,column=0,padx=10,pady=10,sticky="nswe")
+vector_model_builder = tk.Label(master=vmb_frame,text="Vector Model Builder")
+vector_model_builder.grid(row=0,column=0,sticky="w")
+# looping to populate labels
+j=1
+for i, text in enumerate(VMB_ARGS):
+    label = tk.Label(master=vmb_frame,text=VMB_ARGS[i][0])
+    label.grid(row=j,column=0,sticky="e")
+    j+=1
+
+# Argument 1 Dataset
+def open_arg1_browse():
+    """Open a file."""
+    filepath = askopenfilename(
+        filetypes=[("Text Files", "*.txt"),("All Files", "*.*")]
+    )
+    if not filepath:
+        return
+    arg_path_ent.delete("0", tk.END)
+    arg_path_ent.insert(tk.END, filepath)
+arg_path_ent = tk.Entry(master=vmb_frame,width=40)
+arg_path_ent.grid(row=1,column=1,sticky="w")
+arg_path_ent.insert(0,VMB_ARGS[0][1])
+arg1_browse = tk.Button(master=vmb_frame,text="Browse...",command=open_arg1_browse)
+arg1_browse.grid(row=1,column=2,sticky="e")
+
+# Argument 2 Count Method
+method_ent = tk.Entry(master=vmb_frame)
+method_ent.grid(row=2,column=1,sticky="w")
+method_ent.insert(0,VMB_ARGS[1][1])
+
+# Argument 3 Weighting arg
+vmb_weight_var = tk.StringVar(vmb_frame)
+    # add additional options for weighting here
+vmb_weight_var.set(VMB_ARGS[2][1])
+vmb_weight_menu = tk.OptionMenu(vmb_frame, vmb_weight_var,*WEIGHT_METHODS)
+vmb_weight_menu.grid(row=3,column=1,sticky="w")
+
+# Argument 4 Output directory
+def open_outdir_browse():
+    """Open a directory."""
+    dirpath = askdirectory()
+    if not dirpath:
+        return
+    outdir_ent.delete("0", tk.END)
+    outdir_ent.insert(tk.END, dirpath)
+outdir_ent = tk.Entry(master=vmb_frame,width=40)
+outdir_ent.grid(row=4,column=1,sticky="w")
+outdir_ent.insert(0,VMB_ARGS[3][1])
+outdir_browse = tk.Button(master=vmb_frame,text="Browse...",command=open_outdir_browse)
+outdir_browse.grid(row=4,column=2,sticky="e")
+
+# Argument 5 Output File Name
+outf_name_ent = tk.Entry(master=vmb_frame)
+outf_name_ent.grid(row=5,column=1,sticky="w")
+outf_name_ent.insert(0,VMB_ARGS[4][1])
+
+# Argument 6 value of N "-n"
+n_ent = tk.Entry(master=vmb_frame)
+n_ent.grid(row=6,column=1,sticky="w")
+n_ent.insert(0,VMB_ARGS[5][1])
+
+# A dataset is required to run
+def run_vector_model_builder():
+    # Check for the default case of None for output file name
+    outfile_arg = outf_name_ent.get()
+    n_val = n_ent.get()
+    if (outfile_arg == "None"):
+        outfile_arg = None
+    # Convert value of n from string to int or set the default to 3 if blank
+    if (n_val == ""):
```

Comment:
Is it possible for the `get` methods above to return `None`? If so, you could change this to `if not n_val:`, which would cover both `""` and `None`.

Reply:
Get method for outfile_arg, in this case, will always return a string of the name. For this one in particular, it's a string of a name or the string "None". If "None" I have to convert to a bool. Also, I am changing the names to some of these variables and functions to something that should make more sense.